src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -74,6 +99,8 @@ private void ReflectionReadMembers(object obj, XmlReaderDelegator xmlReader, Xml
             bool hasRequiredMembers = (firstRequiredMember < memberCount);
             int requiredIndex = hasRequiredMembers ? firstRequiredMember : -1;
             int index = -1;
+            DataMember[] members = new DataMember[memberCount];
+            ReflectionGetMembers(_classContract, members);
```

Comment:
Should we assert here that the int returned from ReflectionGetMembers is memberCount?  Under what circumstances would it be different?  Are nulls in members[] allowed it they are not all filled in?